lib/editor/components/EditorInput.js

```diff
@@ -314,7 +316,18 @@ export default class EditorInput extends Component<Props> {
                 ? <option disabled value=''>{field.required ? '-- select an option --' : '(optional)' }</option>
                 : null
               }
-              {field.options && field.options.map(o => (<option value={o.value} key={o.value}>{o.text || o.value}</option>))}
+              {field.name === 'route_type' &&
+                <optgroup label={standard}>
+                  {field.options.slice(0, 7).map(o => (
+                    <option value={o.value} key={o.value}>{o.text} ({o.value})</option>))}
+                </optgroup>}
+              {field.name === 'route_type' &&
+                <optgroup label={extended}>
+                  {field.options.slice(8).map(o => (
+                    <option value={o.value} key={o.value}>{o.text} ({o.value})</option>))}
+                </optgroup>}
+              )}
```

Comment:
I would recommend placing this stuff (including lines 303 and 304) into a `_renderRouteTypeOptions` method to encapsulate it a bit more. Once that's done, I would recommend you use a ternary here instead of making the check for field name twice:
```javascript
field.name === 'route_type'
  ? this._renderRouteTypeOptions(options)
  : field.options.map(o => (<option value={o.value} key={o.value}>{o.text || o.value}</option>))
```